### PR TITLE
cleanup ErrorDetails variants and ts definition

### PR DESCRIPTION
**Pull request by kziemianek**
Removes unused `ErrorDetails` variants and adds `VerifyWeb3SignatureFailed` to ts definion - it replaced `VerifySubstrateSignatureFailed` and `VerifyEvmSignatureFailed` in Rust code but we forgot to update ts definition.

### CODE REVIEW

**Review by jonalvarezz — APPROVED** [review]

File: tee-worker/client-api/parachain-api/prepare-build/interfaces/identity/definitions.ts [hunk 1]
```diff
@@ -215,11 +215,8 @@ export default {
                 InvalidIdentity: "Null",
                 WrongWeb2Handle: "Null",
                 UnexpectedMessage: "Null",
-                WrongSignatureType: "Null",
-                VerifySubstrateSignatureFailed: "Null",
-                VerifyEvmSignatureFailed: "Null",
-                RecoverEvmAddressFailed: "Null",
-                Web3NetworkOutOfBounds: "Null",
+                __Unused_WrongSignatureType: "Null",
```
Comment (0xverin) [inline tee-worker/client-api/parachain-api/prepare-build/interfaces/identity/definitions.ts]: Can we use same naming format here?

**Review by 0xverin — APPROVED** [review]

File: tee-worker/client-api/parachain-api/prepare-build/interfaces/identity/definitions.ts (hunk 1 above)
Comment (BillyWooo) [reply tee-worker/client-api/parachain-api/prepare-build/interfaces/identity/definitions.ts]: what do you mean by " naming format "? naming pattern? for example?

Comment (0xverin) [reply tee-worker/client-api/parachain-api/prepare-build/interfaces/identity/definitions.ts]: I mean is to use camelCase: `UnusedWrongSignatureType`

Comment (BillyWooo) [reply tee-worker/client-api/parachain-api/prepare-build/interfaces/identity/definitions.ts]: I think he wanted to emphasis on it.

**Review by BillyWooo — APPROVED** [review]

File: tee-worker/client-api/parachain-api/prepare-build/interfaces/identity/definitions.ts (hunk 1 above)
Comment (kziemianek) [reply tee-worker/client-api/parachain-api/prepare-build/interfaces/identity/definitions.ts]: Simply no: by following `__Unused_` `polkadotjs@api` pattern, the generated code will be able to deserialize Enum with discontinous index while skipping this field in generated type see the generated code (there is no `__Unused_WrongSignatureType`):

```
/** @name ErrorDetail */
export interface ErrorDetail extends Enum {
    readonly isImportError: boolean;
    readonly isUnauthorizedSigner: boolean;
    readonly isStfError: boolean;
    readonly asStfError: Bytes;
    readonly isSendStfRequestFailed: boolean;
    readonly isParseError: boolean;
    readonly isDataProviderError: boolean;
    readonly asDataProviderError: Bytes;
    readonly isInvalidIdentity: boolean;
    readonly isWrongWeb2Handle: boolean;
    readonly isUnexpectedMessage: boolean;
    readonly isVerifyWeb3SignatureFailed: boolean;
    readonly type:
        | "ImportError"
        | "UnauthorizedSigner"
        | "StfError"
        | "SendStfRequestFailed"
        | "ParseError"
        | "DataProviderError"
        | "InvalidIdentity"
        | "WrongWeb2Handle"
        | "UnexpectedMessage"
        | "VerifyWeb3SignatureFailed";
}
```

Enum can be discontinous after removing it's variant while keeping rest variant indexes unchanged - introduced in #2247 (check related issue for more info)

**Pull request merged**